File: Backend/Scheduler/Model/OrderReport/GroupBlock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Scheduler.Model.OrderReport
{
    public class GroupBlock
    {
        public int Id { get; set; }

        public int GroupIndex { get; set; }

        public int OrderBlockId { get; set; }
        public virtual OrderBlock OrderBlock { get; set; }

        public long StartTime { get; set; }

        public long Duration { get; set; }

        public virtual Workshop Workshop { get; set; }
        public int WorkshopId { get; set; }

        public virtual List<DetailsBatchBlock> DetailsBatchBlocks { get; set; }

        public int TransportOperationBlockId { get; set; }
        public virtual TransportOperationBlock TransportOperationBlock { get; set; }
    }
}

File: Frontend/HttpClient/src/app/models/Reporting/GroupBlock.ts
import { DetailsBatchBlock } from './DetailsBatchBlock';
import { Workshop } from '../workshop';
import { TransportOperationBlock } from './transport-operation-block';

export class GroupBlock {
    public id: number;

    public groupIndex: number;

    public workshop: Workshop;

    public startTime: number;

    public duration: number;

    public transportStartTime: number;

    public transportDuration: number;

    public detailsBatchBlocks: DetailsBatchBlock[];

    public transportOperationBlock: TransportOperationBlock;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }

        return Object.assign(
            new GroupBlock(),
            obj,
            {
                detailsBatchBlocks: obj.detailsBatchBlocks ? obj.detailsBatchBlocks.map(g => DetailsBatchBlock.fromJSON(g)) : [],
                transportOperationBlock: obj.transportOperationBlock ? TransportOperationBlock.fromJSON(obj.transportOperationBlock) : null
            }
        );
    }
}
